### ISSUE TITLE
[Bug]: `ignore_messages` is not actually supported inline

### ISSUE THREAD

**Issue by abompard** [rstcheck/rstcheck#167]
### To Dos

- [X] I tested with the [latest released version](https://github.com/rstcheck/rstcheck/releases/latest).

- [X] I have checked the [issues](https://github.com/rstcheck/rstcheck/issues) and think that this is not a duplicate.

- [X] I added a very descriptive title to this issue.

### Example Code (rst)

```rst
.. rstcheck: ignore_messages=Duplicate

Title
=====

Title
=====
```

### Relevant log output

```Shell
WARNING:rstcheck_core.inline_config:Unknown inline config 'ignore_messages' found. Source: 'test.rst' at line 1
test.rst:7: (INFO/1) Duplicate implicit target name: "title".
Error! Issues detected.
```

### Description

According to [the documentation](https://rstcheck.readthedocs.io/en/latest/usage/config/#ignore-specific-error-messages), the `ignore_messages` option should be available as an inline option in a RST file. However, it does not seem to be the case, it is reported as an unknown directive:
```
WARNING:rstcheck_core.inline_config:Unknown inline config 'ignore_messages' found. Source: 'docs/release_notes.rst' at line 5
```
And there is no place in `rstcheck_core/inline_config.py` that handles ignored messages.

### Operating System

Linux

### Operating System Details

Fedora 37

### Python Version

3.11.2

### rstcheck Version

rstcheck CLI Version: 6.1.2, rstcheck-core Version: 1.0.3

**Comment by Cielquan** [comment]
It is true that `ignore_messages` is not supported as an inline option in the code.
But the docs do not state that it is supported. The 4 config values above it state that inline comments are a valid source, but `ignore_messages` only states the CLI option and the config file as valid sources.

**Comment by abompard** [comment]
Right, my bad, I misunderstood the "File:" prefix.

I think it would be a useful addition, though: my release notes are autogenerated and therefore have a lot of duplicated implicit targets. I would love to be able to disable that check for this file only while keeping it for manually written docs.

**Comment by Cielquan** [comment]
Maybe it would be a good idea to change the File prefix to Config-File for clarification.

Back when I made a rewrite/refactor with version 6 I think I wanted to include it as an inline config option, but the implementation would not have been as trivial as the others because of the underlying structure of rstcheck which I kept. Message ignores are evaluated later than e.g. language ignores.

For your problem I see two solutions with the current available version, while the first option should be more practical IMO.
1. invoke rstcheck twice. Once for linting the changelog where you either specify the ignore via cli or an seperate config file which you then specify with `--config`. And the second time for your docs.
2. If you only want to invoke rstcheck once you can also work with multiple config files and use the autodetection. Here I assume you have a docs directory and the changelog file is at the root. You place a config file in your docs directory with the config for your docs files and a config file at the root where you specify the ignore for your changelog. This only works of cause when you do not have other rst files in the root you want to check also.
